src/global.rs

```diff
@@ -4,6 +4,41 @@ use std::mem::MaybeUninit;
 use std::str::FromStr;
 use std::sync::Once;
 
+#[cfg(feature = "global")]
+/// Macro for creating symbols from &'static str. Useful for commonly used symbols known at compile time.
+/// This is faster then GlobalSymbol::from(s) by avoiding mutex contention.
+///
+/// # Examples
+///
+/// ```
+/// use symbol_table::static_symbol;
+/// use symbol_table::GlobalSymbol;
+///
+/// let hello = static_symbol!("hello");
+/// assert_eq!(hello, GlobalSymbol::from("hello"));
+///
+/// // The same symbol is returned on subsequent calls
+/// let hello2 = static_symbol!("hello");
+/// assert_eq!(hello, hello2);
+/// ```
+#[macro_export]
+macro_rules! static_symbol {
+    ($s:expr) => {{
+        use std::mem::MaybeUninit;
+        use std::sync::Once;
+        static mut SYMBOL: MaybeUninit<GlobalSymbol> = MaybeUninit::uninit();
+        static ONCE: Once = Once::new();
+
+        unsafe {
+            ONCE.call_once(|| {
+                SYMBOL.write(GlobalSymbol::from($s));
+            });
+
+            SYMBOL.assume_init_read()
+        }
+    }};
+}
+
```

Comment:
Could this be a `OnceCell` to avoid the unsafe here? Also, maybe restrict the macro to literals (`$s:lit`) since it will only evaluate the expr the first time to avoid confusion.

Reply:
Both fair points, fixed.